core/graph/src/test/java/uk/gov/gchq/gaffer/graph/hook/GraphHookTest.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2016 Crown Copyright
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.gchq.gaffer.graph.hook;
+
+import org.junit.Test;
+import uk.gov.gchq.gaffer.commonutil.StreamUtil;
+import uk.gov.gchq.gaffer.exception.SerialisationException;
+import uk.gov.gchq.gaffer.jsonserialisation.JSONSerialiser;
+
+
+public abstract class GraphHookTest<T> {
```

Comment:
Should we take a look at the hierarchy of test classes we're using throughout Gaffer and come up with something which is reusable in different modules? For example, the ```GraphHookTest.shouldJsonSerialiseAndDeserialise()``` test method should appear everywhere that we test a class (or type of class) which should be serialisable. Is there any value in a hierarchy which features a ```SerialiseableTest<T>``` interface (possibly with a default implementation)?